lib/domain.js

```diff
@@ -207,7 +206,7 @@ Domain.prototype.members = undefined;
 Domain.prototype._errorHandler = function(er) {
   var caught = false;
 
-  if (!util.isPrimitive(er)) {
+  if (typeof er === 'object' && er !== null || typeof er === 'function') {
```

Comment:
```suggestion
  if ((typeof er === 'object' && er !== null) || typeof er === 'function') {
```

Reply:
I just checked and we currently have 648 mixed operators (reported by eslint using `no-mixed-operators`). Most of those are reported in `lib`.
Checking with `no-extra-parens` I find 457 operators having brackets around them (I used options to exclude a few entries that are not operators).

I think it would be best to have a consistent code style here. I guess we do not care about all operators and it's possible to define own operator groups with `no-mixed-operators` (to make sure we only match the ones we care about most). But since we currently have more mixed operators than the other way around, I would like to stick to this. I am fine with introducing a stricter rule for it though.

Reply:
> I just checked and we currently have 648 mixed operators (reported by eslint using `no-mixed-operators`). Most of those are reported in `lib`.

Most of those are also not mixes of `&&` and `||`, but rather other operators with more intuitive precedence rules, so I wouldn’t consider that statistic a good argument here.

Reply:
Since it's a style preference, would you be fine to open a PR to enforce these two groups to use braces? It's possible by defining `groups` with https://eslint.org/docs/rules/no-mixed-operators. I would also do that but I am quite busy at the moment.

Reply:
+1 to adding parens here

Reply:
I also think the suggested parentheses would help here.

@BridgeAR Can you add the parentheses here and then, if this is a nit you either disagree with or one you get more than infrequently, open a separate issue (or PR) to propose one or more possible linting solutions?

Reply:
Linting was added in https://github.com/nodejs/node/pull/29834.

Reply:
Sure enough, lint results flag this. https://ci.nodejs.org/job/node-test-linter/30156/console